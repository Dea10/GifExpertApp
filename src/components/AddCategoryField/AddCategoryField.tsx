import React, { useState, Dispatch, SetStateAction } from 'react';

type AddCategoryFieldProps = {
    setCategories: Dispatch<SetStateAction<string[]>>;
}

const AddCategoryField = ({ setCategories }: AddCategoryFieldProps) => {

    const [inputValue, setInputValue] = useState('');

    const handleOnChange = ({target}: any) => {
        setInputValue(target.value);
    }

    const handleOnSubmit = (e: any) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories((categories: string[]) => [ inputValue, ...categories ]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleOnSubmit }>
            <input 
                type='text'
                value={ inputValue }
                onChange={ handleOnChange }
            />
        </form>
    );
}

export default AddCategoryField;