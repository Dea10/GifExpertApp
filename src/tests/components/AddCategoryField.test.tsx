import React from 'react';
import { shallow } from 'enzyme';
import AddCategoryField from '../../components/AddCategoryField/AddCategoryField';

describe('AddCategoryField tests', () => {

    const setCategories = () => {};

    test('should render without errors', () => {
        const wrapper = shallow(<AddCategoryField setCategories={ setCategories } />)
        expect(wrapper).toMatchSnapshot();
    });
});