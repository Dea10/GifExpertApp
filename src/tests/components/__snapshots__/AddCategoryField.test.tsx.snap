// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategoryField tests should render without errors 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
